src/python-common/ceph/deployment/service_spec.py

```diff
@@ -1229,6 +1229,7 @@ def __init__(self,
                  rgw_bucket_counters_cache: Optional[bool] = False,
                  rgw_bucket_counters_cache_size: Optional[int] = None,
                  generate_cert: bool = False,
+                 disable_multisite_sync_traffic: Optional[bool] = None,
```

Comment:
nit: as it's a bool I think we should default to `False` intead of None in this case

Reply:
I had originally had that. The problem is I need to be able to differentiate between a user who has explicitly set it to `False` and a user who has not set the setting at all. This is because if a user had manually set `rgw_run_sync_thread` to `false` and then upgraded to a version including this patch with `disable_multisite_sync_traffic` defaulting to `False` we'd see that value of `False` and set `rgw_run_sync_thread` to `True`. If we made it default to `True`, we'd have the same issue the other way around where someone who had `rgw_run_sync_thread` set to `False` would have it get altered by cephadm without them having taken action. Making this default to `None` lets us differentiate between explicit user action and just a default value that appeared during an upgrade.